tests/performance/test_mongo_redis_performance.py

```diff
@@ -0,0 +1,197 @@
+        letter_range,
+        alphabet_range,
+        word_link_percentage,
+        letter_link_percentage,
+        seed,
+    ):
+        self.node_type = 'Concept'
+        self.node_range = [int(i) for i in node_range.split('-')]
+        self.word_range = [int(i) for i in word_range.split('-')]
+        self.letter_range = [int(i) for i in letter_range.split('-')]
+        self.alphabet_range = [int(i) for i in alphabet_range.split('-')]
+        self.word_link_percentage = word_link_percentage
+        self.letter_link_percentage = letter_link_percentage
+        self.seed = seed
+        if seed:
+            random.seed(seed)
+
+    @pytest.fixture(scope="class", autouse=True)
+    def database(self):
+        _db_up()
+        yield
+        _db_down()
+
+    @staticmethod
+    def combinatorial_loop(d: Dict[str, Any], f: Callable[[int, int], None], percentage: float):
+        for v in d.values():
+            v = list(v)
+            for i, _ in enumerate(v):
+                for j in range(i + 1, len(v)):
+                    if random.random() > percentage:
+                        continue
+                    f(v[i], v[j])
+
+    @measure
+    def generate_links_word(self, word_dict):
+        links_dict_word = {}
+        word_count = self.word_range[1] - self.word_range[0]
+        for v in word_dict.values():
+            v = list(v)
+            for i in range(0, len(v)):
+                for j in range(i + 1, len(v)):
+                    if random.random() > self.word_link_percentage:
+                        continue
+                    key = f'{min(v[i], v[j])}->{max(v[i], v[j])}'
+                    if key in links_dict_word:
+                        links_dict_word[key] += 1 / word_count
+                    else:
+                        links_dict_word[key] = 1 / word_count
+        return links_dict_word
+
+    @measure
+    def generate_nodes(self, das: DistributedAtomSpace = None):
+        node_list = []
+        node_names = {}
+        word_dict = {}
+        for N in range(*self.node_range):
+            node_list.append({'name': []})
+            word_list = []
+            for W in range(*self.word_range):
+                node_list[N]['name'].append([])
+                node_list[N]['name'][W - self.word_range[0]] = ''.join(
+                    [chr(97 + randint(*self.alphabet_range)) for L in range(*self.letter_range)]
+                )
+                word_list.append(node_list[N]['name'][W - self.word_range[0]])
+            node_list[N]['name'] = ' '.join(node_list[N]['name'])
+            node_list[N]['type'] = self.node_type
+            das.add_node(node_list[N])
+
+            for w in word_list:
+                if w in word_dict:
+                    word_dict[w].add(N)
+                else:
+                    word_dict[w] = set([N])
+
+            if node_list[N]['name'] in node_names:
+                raise ValueError('Exists')
+            node_names[node_list[N]['name']] = 1
```

Comment:
What does it mean to set 1 to node_names[...]?

It seems to me that node_names should be a set instead of a dict.

Reply:
Changed to set